### ISSUE TITLE
New youtube ads not blocked

### ISSUE THREAD

**Issue by MrGamy** [uBlockOrigin/uAssets#7519]
### URL(s) where the issue occurs

```
https://www.youtube.com/watch?v=4ehDRCE1Z38
https://www.youtube.com/watch?v=aaL7owZmbEA
```

### Describe the issue

Certain videos on youtube seem to implement a new way of displaying ads - those ads are not being blocked by uBlock Origin at the Moment. The Channels around LTT seem to use this new method.

### Versions

- Browser/version: Firefox/77.0.1 (64-bit)
- uBlock Origin version: 1.27.10

### Settings

Used lists: Built in +EasyList, EasyPrivacy, Malware Domains & Malware Domain List, Peter Lowe's 
No other additions to default settings.

**Comment by gwarser** [comment]
Updating "uBlock filters" list does not help?

**Comment by MrGamy** [comment]
didn't help yet (tried updating first) and I only noticed it on those two videos so far. Like I said, no Ads on basically any other videos I've watched today

**Comment by Tomek-PL** [comment]
I have the same issue on some videos.
I noticed that the video-ad has a yellow progress bar and cannot be fast forwarded. Once I manage to turn on my video, after scrolling to the beginning, it again activates the video ad with the yellow bar.
It's annoying

**Comment by gwarser** [comment]
Related https://www.reddit.com/r/uBlockOrigin/comments/gx19kt/ads_on_youtube_as_of_today/

**Comment by krystian3w** [comment]
https://github.com/uBlockOrigin/uBlock-issues/issues/1102

**Comment by MrGamy** [comment]
Could you tell me what you need me to do if I encounter another Ad? I've attached two Network Monitor HAR Files and hope this helps.
https://drive.google.com/drive/folders/1R6IyXLNbaKAAM2UN0ntQU1K9EtNT8T49?usp=sharing

**Comment by gwarser** [comment]
If you updated "uBlock filters" in lates ~20 hours and still dont' have "Skip ad" button, add `youtube.com#@##player-ads` to My filters.

**Comment by Tomek-PL** [comment]
I deleted uBlock from addons, reopen firefox and install uBlock again. No ads for over 1 day so I think it's a success

**Comment by llacb47** [comment]
That does not update the filters. To update the filters you must click "Purge all caches" and then "update now".

**Comment by Tomek-PL** [comment]
OK, now I've also done this. Thank you for the advice

**Comment by gwarser** [comment]
Try adding to My filters:

    youtube.com##+js(set, ytInitialPlayerResponse.adPlacements, undefined)

**Comment by MrGamy** [comment]
> 
> 
> Try adding to My filters:
> 
> ```
> youtube.com##+js(set, ytInitialPlayerResponse.adPlacements, undefined)
> ```
> 
> This should block video ads in new tabs.

This one seems to have worked for me. 
```
youtube.com#@##player-ads
```
doesn't seem to do anything. I'll further watch the issue and report back on any new ads popping up :) Thanks!

**Comment by llacb47** [comment]
If someone is still having issues they could try: https://github.com/dhowe/AdNauseam/commit/e6682bbe3eca3774d0b1159d312a84a8c7afcce7

**Comment by krystian3w** [comment]
Very old:

> committed on 28 Apr 2018

https://github.com/PolishFiltersTeam/PolishAnnoyanceFilters/commit/beab5819336ed2611caa812fbc42b306526101ce

**Comment by gwarser** [comment]
OK, `youtube.com##+js(set, ytInitialPlayerResponse.adPlacements, undefined)` caused issues https://github.com/uBlockOrigin/uAssets/issues/7542 and was removed from filter list https://github.com/uBlockOrigin/uAssets/commit/9d7610baf1449021a2974b2569444a0374c4245f

Maybe some other variant of this filter will work.

Replace `undefined` by:
`false`,
`true`,
`null` - similar to `undefined`, may not work,
`noopFunc`,
`trueFunc`,
`falseFunc` or
`''`.

Like this:

    youtube.com##+js(set, ytInitialPlayerResponse.adPlacements, false)

**Comment by gorhill** [comment]
Also, be sure you do not have another similar-purposed content blocker (ABP, AdBlock, AdGuard, etc.) along uBO; using more than one similar-purposed content blocker will **always** cause issues -- uBO is meant to perform optimally on its own.

**Comment by mlindner** [comment]
I'm also having the same issue on Chrome. It displays a grey overlay with a skip button in the lower right corner, but doesn't actually play the ad.

**Comment by gwarser** [comment]
@mlindner `youtube.com##.ytp-ad-module` and `youtube.com##.video-ads` should hide these grey static ads, but when YT decide to show you video ad, you will not be able to skip it.

**Comment by sharpaper** [comment]
Came here to report the same issue (on firefox). Ads are not blocked on youtube. I can also reproduce the issue with the 2 links posted by OP.

**Comment by krystian3w** [comment]
https://github.com/NanoAdblocker/NanoFilters/issues/518

**Comment by krystian3w** [comment]
IMO latest `json-prune` down, I see ads with `/sw.js` blocked (AdNauseam also fail).

**Comment by llacb47** [comment]
This video is also good for testing: `https://www.youtube.com/watch?v=jjwilAja7Lc`

**Comment by gwarser** [comment]
Do you see ads after opening new tab or when navigating in YT (with red progress bar on top)?

**Comment by gwarser** [comment]
1. with new tabs: look into page source and try location initial player configuration object assignment. `ytInitialPlayerResponse = {...adPlacements/playerAds...}` or `widnow[ytInitialPlayerResponse] = {...adPlacements/playerAds...}` or `somethingConfig` then follow it in debugger and try to defuse by set/aopr/aopw/whatever.
2. when navigation inside YT: set `userResourcesLocation` to `https://gist.githubusercontent.com/gorhill/ef7af0cf1f3c9c6d1b57f355ceb4ce40/raw/61969a9fc2f66e5e5938cc5a30d12419d8288316/json-prune.js` and add to My filters:

    youtube.com##+js(json-prune, [].playerResponse.adPlacements [].playerResponse.playerAds playerResponse.adPlacements playerResponse.playerAds adPlacements playerAds)


<details>
<summary>Details</summary>


```javascript
    (function() {
      var setFiller = function() {
        var fillerData = {
          browse: {
            filler: true,
            page: 'browse'
          },
          home: {
            filler: true,
            page: 'home',
            endpoint: {
              commandMetadata: {
                webCommandMetadata: {
                  url: "/",
                  webPageType: "WEB_PAGE_TYPE_BROWSE"
                }
              },
              urlEndpoint: {
                url: "/"
              }
            },
            response: {
              contents: {
                twoColumnBrowseResultsRenderer: {
                  tabs: [{
                    tabRenderer: {
                      selected: true,
                      content: {
                        richGridRenderer: {
                          contents: [],
                          continuations: [{
                            nextContinuationData: {continuation: ''}
                          }]
                        }
                      }
                    }
                  }]
                }
              }
            }
          },
          search: {
            filler: true,
            page: 'search',
            endpoint: {
              commandMetadata: {
                webCommandMetadata: {
                  url: "/results",
                  webPageType: "WEB_PAGE_TYPE_SEARCH"
                }
              },
              urlEndpoint: {
                url: "/results"
              }
            },
            response: {
              contents: {
                twoColumnSearchResultsRenderer: {
                  primaryContents: {
                    sectionListRenderer: {
                      contents: [],
                      subMenu: {
                        searchSubMenuRenderer: {}
                      }
                    }
                  },
                  secondaryContents: {
                    secondarySearchContainerRenderer: {
                      contents: []
                    }
                  }
                }
              }
            }
          },
          watch: {
            filler: true,
            page: 'watch',
            endpoint: {
              commandMetadata: {
                webCommandMetadata: {
                  url: "/watch",
                  webPageType: "WEB_PAGE_TYPE_WATCH"
                }
              },
              urlEndpoint: {
                url: "/watch"
              }
            },
            playerResponse: {
              playabilityStatus: {
                status: 'OK'
              }
            },
            response: {
              contents: {
                twoColumnWatchNextResults: {
                  results: {
                    results: {
                      contents: [{
                        videoPrimaryInfoRenderer: {},
                        videoSecondaryInfoRenderer: {}
                      }]
                    }
                  },
                  secondaryResults: {
                    secondaryResults: {}
                  }
                }
              }
            }
          }
        };

        window.ytcfg.set('FILLER_DATA', fillerData);
      };

        window.getPageData = function() {
          if (window.ytcsi) {window.ytcsi.tick("pr", null, '');}
          var endpoint = null;
            endpoint = {"commandMetadata":{"webCommandMetadata":{"url":"/watch?v=jgliCvYQ4hY","webPageType":"WEB_PAGE_TYPE_WATCH","rootVe":3832}},"urlEndpoint":{"url":"/watch?v=jgliCvYQ4hY"}};
          var data = {
              page: "watch",
              player: ytplayer.config,
              playerResponse: window.ytInitialPlayerResponse,
                url: "\/watch?v=jgliCvYQ4hY",
              response: window.ytInitialData,
              endpoint: endpoint
          };
          return {
            data: data,
            endpoint: endpoint
          };
        };

        if (window.loadDataHook) {
          var pageData = window.getPageData();
          window.loadDataHook(pageData.endpoint, pageData.data);
          window.loadDataHook = null;
        }
        setFiller();

    })();
```



</details>

**Comment by krystian3w** [comment]
- Only at the beginning, 
- video opened in a new card
- Pre-roll ads is charged once in about 60 tests
- no successful loading attempts during viewing

**Comment by Toshiba1127** [comment]
Weird, I don't have this issue yet.

**Comment by krystian3w** [comment]
I had today.

First such 30 seconds and then a white board with a small circle in the corner for about 5 seconds.

**Comment by krystian3w** [comment]
Something new - I can rate ads without watch that ads:

![Screenshot_2020-07-07 Zebo ft Cheatz - DISS NA KLAUDUSIEK](https://user-images.githubusercontent.com/35370833/86772930-3c9e9380-c055-11ea-9393-07cff83c6757.png) |
--- |

<details><summary> html from devtools </summary>

```html
<paper-dialog role="dialog" tabindex="-1" style="outline: currentcolor none medium; position: fixed; inset: auto 0px 0px auto; box-sizing: border-box; max-height: 147.6px; max-width: 400px; z-index: 2202;" class="style-scope ytd-popup-container"><!--css-build:shady-->
    <ytd-single-option-survey-renderer dialog="true" class="style-scope ytd-popup-container" tabindex="-1"><!--css-build:shady-->
    
    <div id="header" class="style-scope ytd-single-option-survey-renderer">
      <yt-formatted-string class="style-scope ytd-single-option-survey-renderer" aria-label="Oceń reklamy, które wyświetliły Ci się ostatnio w YouTube.">Oceń reklamy, które wyświetliły Ci się ostatnio w YouTube.</yt-formatted-string>
      <yt-icon-button class="style-scope ytd-single-option-survey-renderer"><!--css-build:shady--><button id="button" class="style-scope yt-icon-button" aria-label="Zamknij">
        <yt-icon icon="yt-icons:close" class="style-scope ytd-single-option-survey-renderer"><svg viewBox="0 0 24 24" preserveAspectRatio="xMidYMid meet" focusable="false" style="pointer-events: none; display: block; width: 100%; height: 100%;" class="style-scope yt-icon"><g class="style-scope yt-icon">
        <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z" class="style-scope yt-icon"></path>
      </g></svg><!--css-build:shady-->
    
    
  </yt-icon>
      </button></yt-icon-button>
    </div>
    <div id="options" class="style-scope ytd-single-option-survey-renderer"><ytd-single-option-survey-option-renderer class="style-scope ytd-single-option-survey-renderer"><!--css-build:shady-->
    
    <yt-icon-button class="style-scope ytd-single-option-survey-option-renderer"><!--css-build:shady--><button id="button" class="style-scope yt-icon-button">
      <yt-icon class="style-scope ytd-single-option-survey-option-renderer"><svg viewBox="0 0 24 24" preserveAspectRatio="xMidYMid meet" focusable="false" style="pointer-events: none; display: block; width: 100%; height: 100%;" class="style-scope yt-icon"><g class="style-scope yt-icon">
        <path d="M11.99 2C6.47 2 2 6.47 2 12s4.47 10 9.99 10S22 17.53 22 12 17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm4.18-12.24l-1.06 1.06-1.06-1.06L13 8.82l1.06 1.06L13 10.94 14.06 12l1.06-1.06L16.18 12l1.06-1.06-1.06-1.06 1.06-1.06zM7.82 12l1.06-1.06L9.94 12 11 10.94 9.94 9.88 11 8.82 9.94 7.76 8.88 8.82 7.82 7.76 6.76 8.82l1.06 1.06-1.06 1.06zM12 14c-2.33 0-4.31 1.46-5.11 3.5h10.22c-.8-2.04-2.78-3.5-5.11-3.5z" class="style-scope yt-icon"></path>
        <path fill="none" d="M0 0h24v24H0z" class="style-scope yt-icon"></path>
      </g></svg><!--css-build:shady-->
    
    
  </yt-icon>
    </button></yt-icon-button>
    <yt-formatted-string class="style-scope ytd-single-option-survey-option-renderer">Bardzo zły</yt-formatted-string>
  </ytd-single-option-survey-option-renderer><ytd-single-option-survey-option-renderer class="style-scope ytd-single-option-survey-renderer"><!--css-build:shady-->
    
    <yt-icon-button class="style-scope ytd-single-option-survey-option-renderer"><!--css-build:shady--><button id="button" class="style-scope yt-icon-button">
      <yt-icon class="style-scope ytd-single-option-survey-option-renderer"><svg viewBox="0 0 24 24" preserveAspectRatio="xMidYMid meet" focusable="false" style="pointer-events: none; display: block; width: 100%; height: 100%;" class="style-scope yt-icon"><g class="style-scope yt-icon">
        <path fill="none" d="M0 0h24v24H0z" class="style-scope yt-icon"></path>
        <circle cx="15.5" cy="9.5" r="1.5" class="style-scope yt-icon"></circle>
        <circle cx="8.5" cy="9.5" r="1.5" class="style-scope yt-icon"></circle>
        <path d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm0-6c-2.33 0-4.32 1.45-5.12 3.5h1.67c.69-1.19 1.97-2 3.45-2s2.75.81 3.45 2h1.67c-.8-2.05-2.79-3.5-5.12-3.5z" class="style-scope yt-icon"></path>
      </g></svg><!--css-build:shady-->
    
    
  </yt-icon>
    </button></yt-icon-button>
    <yt-formatted-string class="style-scope ytd-single-option-survey-option-renderer">Słaby</yt-formatted-string>
  </ytd-single-option-survey-option-renderer><ytd-single-option-survey-option-renderer class="style-scope ytd-single-option-survey-renderer"><!--css-build:shady-->
    
    <yt-icon-button class="style-scope ytd-single-option-survey-option-renderer"><!--css-build:shady--><button id="button" class="style-scope yt-icon-button">
      <yt-icon class="style-scope ytd-single-option-survey-option-renderer"><svg viewBox="0 0 24 24" preserveAspectRatio="xMidYMid meet" focusable="false" style="pointer-events: none; display: block; width: 100%; height: 100%;" class="style-scope yt-icon"><g class="style-scope yt-icon">
        <path d="M9 14h6v1.5H9z" class="style-scope yt-icon"></path>
        <circle cx="15.5" cy="9.5" r="1.5" class="style-scope yt-icon"></circle>
        <circle cx="8.5" cy="9.5" r="1.5" class="style-scope yt-icon"></circle>
        <path d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z" class="style-scope yt-icon"></path>
        <path fill="none" d="M0 0h24v24H0z" class="style-scope yt-icon"></path>
      </g></svg><!--css-build:shady-->
    
    
  </yt-icon>
    </button></yt-icon-button>
    <yt-formatted-string class="style-scope ytd-single-option-survey-option-renderer">Średni</yt-formatted-string>
  </ytd-single-option-survey-option-renderer><ytd-single-option-survey-option-renderer class="style-scope ytd-single-option-survey-renderer"><!--css-build:shady-->
    
    <yt-icon-button class="style-scope ytd-single-option-survey-option-renderer"><!--css-build:shady--><button id="button" class="style-scope yt-icon-button">
      <yt-icon class="style-scope ytd-single-option-survey-option-renderer"><svg viewBox="0 0 24 24" preserveAspectRatio="xMidYMid meet" focusable="false" style="pointer-events: none; display: block; width: 100%; height: 100%;" class="style-scope yt-icon"><g class="style-scope yt-icon">
        <path fill="none" d="M0 0h24v24H0z" class="style-scope yt-icon"></path>
        <circle cx="15.5" cy="9.5" r="1.5" class="style-scope yt-icon"></circle>
        <circle cx="8.5" cy="9.5" r="1.5" class="style-scope yt-icon"></circle>
        <path d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm0-4c-1.48 0-2.75-.81-3.45-2H6.88c.8 2.05 2.79 3.5 5.12 3.5s4.32-1.45 5.12-3.5h-1.67c-.7 1.19-1.97 2-3.45 2z" class="style-scope yt-icon"></path>
      </g></svg><!--css-build:shady-->
    
    
  </yt-icon>
    </button></yt-icon-button>
    <yt-formatted-string class="style-scope ytd-single-option-survey-option-renderer">Dobry</yt-formatted-string>
  </ytd-single-option-survey-option-renderer><ytd-single-option-survey-option-renderer class="style-scope ytd-single-option-survey-renderer"><!--css-build:shady-->
    
    <yt-icon-button class="style-scope ytd-single-option-survey-option-renderer"><!--css-build:shady--><button id="button" class="style-scope yt-icon-button">
      <yt-icon class="style-scope ytd-single-option-survey-option-renderer"><svg viewBox="0 0 24 24" preserveAspectRatio="xMidYMid meet" focusable="false" style="pointer-events: none; display: block; width: 100%; height: 100%;" class="style-scope yt-icon"><g class="style-scope yt-icon">
        <path d="M11.99 2C6.47 2 2 6.47 2 12s4.47 10 9.99 10S22 17.53 22 12 17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm1-10.06L14.06 11l1.06-1.06L16.18 11l1.06-1.06-2.12-2.12zm-4.12 0L9.94 11 11 9.94 8.88 7.82 6.76 9.94 7.82 11zM12 17.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z" class="style-scope yt-icon"></path>
        <path fill="none" d="M0 0h24v24H0z" class="style-scope yt-icon"></path>
      </g></svg><!--css-build:shady-->
    
    
  </yt-icon>
    </button></yt-icon-button>
    <yt-formatted-string class="style-scope ytd-single-option-survey-option-renderer">Bardzo dobry</yt-formatted-string>
  </ytd-single-option-survey-option-renderer></div>
  </ytd-single-option-survey-renderer>
</paper-dialog>
```

</details>

@ryanbr  / @Khrin:

`youtube.com##ytd-single-option-survey-renderer`

❓

**Comment by Bodudan** [comment]
In short this can help against video ads:
`||www.youtube.com/get_video_info*=adunit&$important`

TL DR:
Setup: FF 56 x86 on Win 7 x64
Hit today ads on some videos on one of machines. After some testing found that:
1) if "uBlock filters" list disabled, ads dont shown
2) from comparision of network activity with and without this list, turned out that rule `@@||youtube.com/get_video_info` overrides rule `=adunit&` from EasyList
3) adding beforementioned static rule removed ads

TO DO:
test that this does not break anything useful

**Comment by krystian3w** [comment]
Help or you have only 1 pre-roll ad at day as to overcome advertising filters.

**Comment by gwarser** [comment]
> In short this can help against video ads:
> `||www.youtube.com/get_video_info*=adunit&$important`

We've been there. You are lucky, for other people this causes "error occured" screens.

**Comment by AntiSol** [comment]
I've also been seeing ads on youtube every ~5th video I try to watch for a couple of weeks now. I've tried many many things.

This:
````||www.youtube.com/get_video_info*=adunit&$important````
seems to work for me, thanks Bodudan. I can't say for sure that it works 100% but I haven't seen any ads in about 10 views since I added it. It doesn't seem to break anything for me.

**Comment by gorhill** [comment]
> for other people this causes "error occurred" screens.

Betweem "error occurred" and video ads not blocked, isn't the former preferable -- at least uBO would be worst than other blockers?

**Comment by okiehsch** [comment]
Adguard uses 
`||youtube.com/get_video_info?*=adunit&$important`
in it's filter list.
I can't reproduce ads or the "error occurred" message on my end.

**Comment by gorhill** [comment]
I have added the filter, let's see the feedback.

**Comment by ryanbr** [comment]
So we should be removed in Easylist @gorhill  ?

**Comment by gwarser** [comment]
||youtube.com/get_video_info?*=adunit&$important

This is not everything. You should also block `||youtube.com/api/stats/watchtime?*=adunit&$important`, which is allowed in EasyPrivacy because people complained about watching time not being saved.

**Comment by okiehsch** [comment]
> So what should be removed in Easylist 

The new filter overwrites our own exception filter.
I do not think any EasyList filter can be removed.

**Comment by gorhill** [comment]
I am just trying to act according to feedback above:

- Two users report that the filter appears to prevent ads from being shown -- so far; hoping for more follow-up feedback;
- @okiehsch says AdGuard uses that filter, and report no ads and no "error occurred"
- Your `json-prune` filter is something which didn't exist originally, so not really full circle

For me the worst case is people being served ads which would not being served without our uBO-specific exception filters.

**Comment by okiehsch** [comment]
> - @okiehsch says AdGuard uses that filter, and report no ads and no "error occurred"

Just to be clear, I can't reproduce the ads no matter what.
I have not seen any youtube ads since this whole discussion started and I use it regularly.

**Comment by gorhill** [comment]
> I have not seen any youtube ads since this whole discussion started and I use it regularly.

Ok, me neither, but since AdGuard uses the filter, surely there must be a reason, for now I haven't taken the time to track down the reason they added the filter.

**Comment by ryanbr** [comment]
If there changes we need to make in Easylist or Easyprivacy, to counter Ads. We'll need to do.  Just need to be safe it doesn't cause the `Error Occurred` issues

**Comment by gwarser** [comment]
> I am just trying to act according to feedback above:
> 
> * Two users report that the filter appears to prevent ads from being shown -- so far; hoping for more follow-up feedback;

And one uses Fx 56. @BogudanPro are you on uBO legacy?

> * @okiehsch says AdGuard uses that filter, and report no ads and no "error occurred"

Tons of issues on AdGuard tracker (it's hard to track them all), but AFAIK uBO does not have problems with "error occured" (uBO can have it for few seconds max) and they still have.

> * Your `json-prune` filter is something which didn't exist originally, so not really full circle

Without `/get_video_info` there will be nothing to prune. This JSON is included in this request and sometimes coincidentally also valid video info and this causes issues.

Hopefully 1.28 with wildcards will fix more, but still it only fixes cases where ad info is downloaded by XHR - this is probably only the case when people navigate to video inside YT page.

> For me the worst case is people being served ads which would not being served without our uBO-specific exception filter, `@@||youtube.com/get_video_info`.

Or complain about not seeing videos at all.
 
> We don't need to block `||youtube.com/api/stats/watchtime?*=adunit&$important`, you say it was already excepted by EasyPrivacy, so its not uBO-specific.

What I meant, some people don't have `youtube.com/get_video_info` request and still see ads because ad info can be smuggled in other requests including `youtube.com/api/stats/watchtime`.

**Comment by gorhill** [comment]
Ok well you have been dealing with the Youtube issue more than I did, so it's best for me to not interfere at least until I invest as much time in this as you did. Feel free to remove the filter I added if you think it makes things worst, I don't have the same insights as you have. It seems what's really missing at this point is users suffering the ads and with minimal technical skills thoroughly collaborating with us to get to the bottom of this.

**Comment by gwarser** [comment]
I will wait for feedback. There is always a chance I was mislead or lost in contradicting cases.

>It seems what's really missing at this point is users suffering the ads and with minimal technical skills thoroughly collaborating with us to get to the bottom of this.

:+1:

**Comment by BogudanPro** [comment]
> And one uses Fx 56. @BogudanPro are you on uBO legacy?
@gwarser, on home pc: yes, uBO 1.16.4.18 legacy. Not sure about PC where i saw ads...

**Comment by AntiSol** [comment]
To chime in with some more info in case it matters: I'm using uBlock Origin v1.18.6 on waterfox classic 2020.06.

>thoroughly collaborating with us to get to the bottom of this

I'd love to help solve this in any way that I can. What should I do if I do see an ad? Is there some info I can collect to help you track this down? Thanks! :)

**Comment by Yuki2718** [comment]
> 
> for now I haven't taken the time to track down the reason they added the filter.

FYI: https://github.com/AdguardTeam/AdguardFilters/commit/2d4f91e7eb55cb87c2a1ae38a5931394cdf1e511

**Comment by BogudanPro** [comment]
Then think it's this 3 issues:
https://github.com/AdguardTeam/AdguardFilters/issues/54573
https://github.com/AdguardTeam/AdguardFilters/issues/55930
https://github.com/AdguardTeam/AdguardFilters/issues/55978

**Comment by gwarser** [comment]
> yes, uBO 1.16.4.18 legacy.

`json-prune` should be available, yet you are few versions behing.

**Comment by gwarser** [comment]
> I'm using uBlock Origin v1.18.6 on waterfox classic 2020.06.

`json-prune` is available starting with 1.23 and legacy 1.16.4.18.

**Comment by sharpaper** [comment]
Is there any fix for this issue? Youtube ads not blocked.

**Comment by mapx-** [comment]
> Is there any fix for this issue? Youtube ads not blocked.

did you update uBO filters ?

**Comment by ryanbr** [comment]
Is it fixed with latest update @sharpaper ?

**Comment by twisterniq** [comment]
Updating uBO filters doesn't help, still have the same issue. On many videos YouTube Ads aren't blocked.

1.27.10
Google Chrome 83.0.4103.116

**Comment by guft** [comment]
I'm getting ads on youtube (Firefox 78.0.2 64bit, uBlock Origin 1.28.0). I can reproduce the issue with the links provided. Just tried on Google Chrome ( 64bit, v83.0.4103.116, uBlock 1.27.10) and can also reproduce.

**Comment by sharpaper** [comment]
@ryanbr Under "Filter lists" I've clicked "Purge caches" and then "Update now". Still seeing ads.

**Comment by AntiSol** [comment]
I'm now very confident that 

````
||www.youtube.com/get_video_info*=adunit&$important
````
is working great. I've watched maybe 50 youtube videos since adding it. No ads, no errors.

**Comment by sharpaper** [comment]
To be fair, I do have noticed a lot fewer ads. The links reported in the ticket do not show any ad anymore. Occasionally (rarely) one shows up though, or I see a 15-seconds whitescreen with the "Skip ad >" button. But overall I think it's working again. Thanks for fixing this.

**Comment by BogudanPro** [comment]
On computer where i've seen ads was FF 56 with uBO 1.13 legacy. As i mentioned before, rule helped to get rid of ads.

**Comment by krystian3w** [comment]
So use too old version, try  override **Firefox add-on signing requirement** (verification):

[https://web.archive.org/web/20180809075244/**http://www.largrizzly.net/firefox.html**](https://web.archive.org/web/20180809075244/http://www.largrizzly.net/firefox.html)

or

`https://www.ghacks.net/2016/08/14/override-firefox-add-on-signing-requirement/#comment-4291219`

If any people don't try publish compatible `json-prune` with uBO 1.13: https://github.com/uBlockOrigin/uAssets/issues/7519#issuecomment-646890235

**Comment by twisterniq** [comment]
Actually, at the moment, the difference is that there is a white screen with "Skip ad" button. Should it be like it's now? I mean, before there were no white screens, no ads, we could directly watch the video.

![image](https://user-images.githubusercontent.com/31934972/87428052-0f685d00-c5e2-11ea-9c2e-ceb9d285d6de.png)

**Comment by Toastbroti** [comment]
`youtube.com##+js(set, ytInitialPlayerResponse.adPlacements, undefined)`

fixes these blank ads without causing any issues for me.

**Comment by twisterniq** [comment]
I can confirm there is no more white screen with the filter given by @Toastbroti. It works fine for me.

**Comment by krystian3w** [comment]
And no break now home page (only blank boxes on background)?

**Comment by Toastbroti** [comment]
> And no break now homepage (only blank boxes on background)?

It doesn't cause any issues

**Comment by lain566** [comment]
> youtube.com##+js(set, ytInitialPlayerResponse.adPlacements, undefined)
> 
> fixes these blank ads without causing any issues for me.

Here's why it was removed from the filter list https://github.com/uBlockOrigin/uAssets/issues/7519#issuecomment-641628768
https://www.reddit.com/r/uBlockOrigin/comments/gzxzv4/youtube_on_firefox_broken/



Some people are still complaining
https://www.reddit.com/r/uBlockOrigin/comments/hq1rgl/getting_yt_ads_suddenly/fxygaes/

**Comment by mapx-** [comment]
see YT issues
https://github.com/uBlockOrigin/uAssets/issues/7636
